messaging/src/main/java/org/axonframework/serialization/upcasting/event/InitialEventRepresentation.java

```diff
@@ -148,4 +148,9 @@ public Instant getTimestamp() {
     public LazyDeserializingObject<MetaData> getMetaData() {
         return metaData;
     }
+
+    @Override
+    public <D> boolean canConvertDataTo(Class<D> requiredType) {
```

Comment:
This is getting really nitty gritty (sorry for that), but the explicit generic parameter (`<D>` doesn't seem to add much value. Perhaps just using `Class<?> expectedType` is enough?

Reply:
Thank you @abuijze!

You are completely right and I'm happy you take it seriously with every bit that gets into the product. This is true quality, since it is not just your habit or a personal style but is technically reasonable and correct.

So I rather get a review with five corrections than an accept of bogus code...

I've just corrected it and pushed....